protostar/commands/test/cheatcodes/mock_call_cheatcode.py

```diff
@@ -0,0 +1,126 @@
+from collections.abc import Mapping
+from typing import TYPE_CHECKING, Dict, List, Type, Union
+
+from starkware.cairo.lang.vm.relocatable import RelocatableValue
+from starkware.starknet.business_logic.execution.execute_entry_point_base import (
+    ExecuteEntryPointBase,
+)
+from starkware.starknet.business_logic.execution.objects import (
+    TransactionExecutionContext,
+)
+from starkware.starknet.public.abi import get_selector_from_name
+from starkware.starknet.storage.starknet_storage import BusinessLogicStarknetStorage
+
+from protostar.commands.test.starkware.cheatable_starknet_general_config import (
+    CheatableStarknetGeneralConfig,
+)
+from protostar.commands.test.starkware.cheatable_syscall_handler import (
+    CheatableSysCallHandler,
+)
+from protostar.commands.test.test_environment_exceptions import CheatcodeException
+from protostar.utils.data_transformer_facade import DataTransformerFacade
+from protostar.utils.starknet_compilation import StarknetCompiler
+
+if TYPE_CHECKING:
+    from protostar.commands.test.starkware.cheatable_state import CheatableCarriedState
+
+
+class MockCallCheatcode(CheatableSysCallHandler):
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        execute_entry_point_cls: Type[ExecuteEntryPointBase],
+        tx_execution_context: TransactionExecutionContext,
+        state: "CheatableCarriedState",
+        caller_address: int,
+        contract_address: int,
+        starknet_storage: BusinessLogicStarknetStorage,
+        general_config: CheatableStarknetGeneralConfig,
+        initial_syscall_ptr: RelocatableValue,
+    ):
+        super().__init__(
+            execute_entry_point_cls=execute_entry_point_cls,
+            tx_execution_context=tx_execution_context,
+            state=state,
+            caller_address=caller_address,
+            contract_address=contract_address,
+            starknet_storage=starknet_storage,
+            general_config=general_config,
+            initial_syscall_ptr=initial_syscall_ptr,
+        )
+        self.data_transformer_facade = DataTransformerFacade(
+            StarknetCompiler(
+                include_paths=general_config.cheatcodes_cairo_path,
+                disable_hint_validation=True,
```

Comment:
Should we disable hint validation here by default?

Reply:
Nah, `StarknetCompiler` shouldn't be strict, at least in this place.

Reply:
I think it sould be consistent with compilation in `declare` cheatcode, #382 We can revisit it working on this one to ensure consistency. Right now leave it whatever state you want.

Reply:
> I think everythink it's okay, but we have to adjust it to refactored cheatcodes.
> 
> Also what is current status for, there is no hurry, just asking because I feel a bit lost in all of the changes:
> 
> * `prepare` transformer
> * `deploy_contract` transformer
> * `expect_events`  transformer

I think `deploy_contract` stopped supporting data transformation. I am also lost with that. We shouldn't refactor cheatcodes in this release. Too many changes everywhere. The cheatcode refactor didn't include `deploy_contract`. I need to recreate it. Arrgsgsdgjdf;lghjkldfgjhoai;t

Prepare and expect_events doesn't support data transformation.

Reply:
Updated mock call to use the refactored approach.
Added data transformer to `deploy_contract`, and prepare cheatcodes.